### PR TITLE
[WIP] add :ucarp label to ucarp address on CentOS

### PR DESCRIPTION
**Pull request by jyotty**
Ubuntu's ucarp implementation automatically does this, and some of our tooling is dependent on it.

### CODE REVIEW

File: roles/neutron-data-network/tasks/main.yml [hunk 1]
```diff
@@ -25,6 +25,18 @@
     name: iputils
   when: os == 'rhel'
 
+- name: install custom ucarp scripts - rhel
+  template:
+    src: "usr/local/libexec/{{ item }}"
```
Comment (scoop206) [inline roles/neutron-data-network/tasks/main.yml]: /usr/local/libexec/ucarp-vip-{up,down} is different than what package places ?

Comment (scoop206) [inline roles/neutron-data-network/tasks/main.yml]: ah, here says it's different (nice):
https://www.rpmfind.net/linux/RPM/fedora/24/x86_64/u/ucarp-1.5.2-15.fc24.x86_64.html

Do we need to create /usr/local/libexec dir ?

Comment (jyotty) [inline roles/neutron-data-network/tasks/main.yml]: Yep, all it adds is a `label` argument to `ip addr add/del` so it shows up in `ip addr show` with `:ucarp` as it does on Ubuntu. (Using a label with delete is a no-op; just there for symmetry)

File: roles/common/tasks/networking.yml [hunk 2]
```diff
@@ -13,6 +13,24 @@
...
+    owner: root
+    mode: 0600
+  when: ucarp is defined and os == 'rhel'
+  tags: ['prereboot']
+
+- name: enable ucarp@{{ "%03d" | format(ucarp.vid) }} service
+# systemd: # Ansible 2.2+ only
+#   name: ucarp@{{ "%03d" | format(ucarp.vid) }}
+#   enabled: yes
+  command: systemctl enable ucarp@{{ "%03d" | format(ucarp.vid) }}
```
Comment (jyotty) [inline roles/common/tasks/networking.yml]: Irksome that the jinja2 `format()` filter is [old-style sprintf formatting](http://jinja.pocoo.org/docs/dev/templates/#format), and not at all what `"string".format()` does in python

**Comment by paulczar** [comment]
:+1:

**Comment by paulczar** [comment]
retest

**Comment by ssrikant20** [comment]
Why do we add ucarp definition to ursula? Shouldn't this be part of ursula-flotsam ?
ursula is a public repository and not all customer install will need ucarp definition written.

**Comment by jyotty** [comment]
@ssrikant20 ucarp has to be configured at the very first stage (network setup). It belongs here.

**Comment by scoop206** [comment]
@ssrikant20  is correct in that is how we did it before.

The ucarp attrs are placed /etc/network/interfaces
The boostrap playbook of flotsam writes the vip-up and vip-down scipts
Basically, ucarp is fully functional after you bootstrap a host.

With RHOSP we could probably actually do it in ursula if that makes sense as I don't think there is anything that actually requires it to be done in bootstrap.

**Comment by ssrikant20** [comment]
After installing ucarp package OS reboot is required to correctly bring up the interface. We do not reboot OS in ursula. When bootstrap.yml was part of ursula, we would reboot the OS to enable ucarp, not any more.

**Pull request merged**